pkg/operator/options/options_validation.go

```diff
@@ -0,0 +1,67 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+
+	"go.uber.org/multierr"
+)
+
+func (o Options) Validate() error {
+	return multierr.Combine(
+		o.validateEndpoint(),
+		o.validateClusterName(),
+		o.validateVMMemoryOverheadPercent(),
+		o.validateAssumeRoleDuration(),
+	)
+}
+
+func (o Options) validateAssumeRoleDuration() error {
+	if o.AssumeRoleDuration < time.Minute*15 {
+		return fmt.Errorf("assume-role-duration cannot be less than 15 minutes")
+	}
+	return nil
+}
+
+func (o Options) validateClusterName() error {
+	// TODO: Remove with karpenter-global-settings
+	if !o.setFlags["cluster-name"] {
+		return nil
+	}
+	if o.ClusterName == "" {
```

Comment:
We shouldn't fail here, right? Since it can be set in either place?

Reply:
That's why we return `nil` if the cluster name wasn't set. The only case it would fail here is if someone explicitly set cluster name to `""`. If they don't specify it as a cli flag or env var it will fall back to settings.

Reply:
Discussed offline, while this behavior was correct it didn't need to happen both pre and post merge. For now required field checks will only take place post merge with settings.